### PR TITLE
Don't try to create tunnel when tiller is outdated

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#8479

### CODE REVIEW

File: helmclient.go [hunk 1]
```diff
@@ -698,7 +700,7 @@ func (c *Client) newTunnel() (*k8sportforward.Tunnel, error) {
 	err = validateTillerVersion(pod, c.tillerImage)
 	if err != nil {
 		if IsTillerInvalidVersion(err) && !c.tillerUpgradeEnabled {
-			c.logger.Log("level", "debug", "message", "found an out-dated tiller but keep going to create a tunnel")
+			return nil, microerror.Maskf(tillerOutdatedError, "found an out-dated tiller but upgrades not enabled")
```
Comment (rossf7) [inline helmclient.go]: Tiller is upgraded by chart-operator in the tenant cluster. When we want to upgrade Tiller we deploy a new version of chart-operator.

So when upgrades are disabled in app-operator we should return the error and cancel the resource.

**Comment by rossf7** [comment]
Tested on `ginger` and with `asgard/f24sd`.

**Comment by rossf7** [comment]
Going with this to get app-operator problems in asgard sorted.

**Comment by xh3b4sd** [comment]
I see we fixed a lot of bugs in the past weeks here. Is there a light at the end of the tunnel or will this go on for some time?

**Comment by rossf7** [comment]
The current helmclient has many issues due to Tiller and the crappy port forwarding connection. So I'm just fixing the high priority bugs.

This is why the Helm 3 upgrade involves a rewrite of helmclient. I hope we can learn from some of the mistakes we made the first time.

The spec is here. I'd love another couple of approvals before merging.

https://github.com/giantswarm/giantswarm/pull/8458

**Pull request merged**